packet/endpoints_test.go

```diff
@@ -92,3 +94,54 @@ func TestDiscoverHardwareFromDHCP(t *testing.T) {
 		})
 	}
 }
+
+func TestGetWorkflowsFromTink(t *testing.T) {
+	ht := &httptest.Server{URL: "FakeURL"}
+	u, err := url.Parse(ht.URL)
+	if err != nil {
+		t.Fatal(err)
+	}
+	c := NewMockClient(u)
+	var tests = []struct {
+		hwID string
+		wcl  *tw.WorkflowContextList
+		err  error
+		c    bool
+	}{
+		{hwID: "Hardware-fake-bde9-812726eff314",
+			wcl: &tw.WorkflowContextList{
+				WorkflowContexts: []*tw.WorkflowContext{
+					&tw.WorkflowContext{
+						WorkflowId:         "active-fake-workflow-bde9-812726eff314",
+						CurrentActionState: 0,
+					},
+				},
+			},
+			err: nil,
+			c:   true,
+		},
+		{hwID: "Hardware-fake-bde9-NoClient",
+			wcl: &tw.WorkflowContextList{},
+			err: errors.New("unable to fetch workflow, client does not exist"),
+			c:   false,
+		},
+	}
+	for _, hw := range tests {
+		if hw.c == false {
+			c.workflowClient = nil
+		} else {
+			ctrl := gomock.NewController(t)
+			defer ctrl.Finish()
+
+			cMock := workflowMock.NewMockWorkflowSvcClient(ctrl)
+			cMock.EXPECT().GetWorkflowContexts(gomock.Any(), gomock.Any()).Return(hw.wcl, hw.err)
+			c.workflowClient = cMock
+		}
+		w, err := c.GetWorkflowsFromTink(hw.hwID)
+		if hw.err != nil {
+			assert.Error(t, err)
+			continue
+		}
+		assert.Equal(t, w, hw.wcl)
```

Comment:
please wrap in a `t.Run(...)` subtest call. This way if a test fails we get better info, and can even run just that test when doing `-run`

Reply:
done,